tests/backup/backup_restore_basic_test.go

```diff
@@ -1068,3 +1070,142 @@ var _ = Describe("{AllNSBackupWithIncludeNewNSOption}", func() {
+		backupLocationMap  map[string]string
+		backupClusterName  string
+		srcVersion         semver.Version
+		destVersion        semver.Version
+		clusterUid         string
+		clusterStatus      api.ClusterInfo_StatusInfo_Status
+	)
+	namespaceMapping := make(map[string]string)
+	duplicateBackupNameMap := make(map[string]string)
+	JustBeforeEach(func() {
+		StartTorpedoTest("BackupRestoreOnDifferentK8sVersions", "Restore from duplicate backup on cluster with different kubernetes version", nil, 83721)
+		log.InfoD("Scheduling applications")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts := ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				log.Infof("Scheduled application with namespace [%s]", namespace)
+				appNamespaces = append(appNamespaces, namespace)
+			}
+		}
+	})
+	It("Validate restore from duplicate backup on cluster with different kubernetes version", func() {
+		Step("Validate applications", func() {
+			log.InfoD("Validating applications")
+			ValidateApplications(contexts)
+		})
+		Step("Configure source and destination clusters with px-central-admin", func() {
+			log.InfoD("Configuring source and destination clusters with px-central-admin")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			dash.VerifyFatal(err, nil, "Fetching px-central-admin ctx")
+			err = CreateSourceAndDestClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of source [%s] and destination [%s] clusters with px-central-admin ctx", SourceClusterName, destinationClusterName))
+			backupClusterName = SourceClusterName
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, backupClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying registration of cluster named [%s] with uid [%s] as backup cluster", backupClusterName, clusterUid))
+		})
+		Step("Fetching source cluster kubernetes version", func() {
+			log.InfoD("Fetching source cluster kubernetes version")
+			version, err := k8s.ClusterVersion()
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching source clutser version %v", version))
+			srcVersion, err = semver.Make(version)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching source clutser version %v", srcVersion))
+		})
+		Step("Fetching destination cluster kubernetes version", func() {
+			log.InfoD("Switching context to destination cluster")
+			SetDestinationKubeConfig()
+			version, err := k8s.ClusterVersion()
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching destination clutser version %v", version))
+			destVersion, err = semver.Make(version)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching destination clutser version %v", destVersion))
```

Comment:
Here we can use "log.FailOnError" for destVersion, err = semver.Make(version) instead of dash.VerifyFatal to avoid too many log statements